sdk/batch/Azure.Compute.Batch/README.md

```diff
@@ -0,0 +1,442 @@
+
+`GetNodeAsync` can be used to retrieve an allocated `BatchNode` from a pool.
+
+```C# Snippet:Batch_Readme_NodeRetreivalAsync
+BatchNode batchNode = await _batchClient.GetNodeAsync("<poolId>", "<nodeId>");
+Console.WriteLine(batchNode.Id);
+Console.WriteLine(batchNode.Url);
+Console.WriteLine(batchNode.State);
+```
+
+### List Nodes asynchronously
+
+`GetNodesAsync` can be used to list all `BatchNode` allocated under a pool
+
+```C# Snippet:Batch_Readme_ListNodesAsync
+await foreach (BatchNode item in _batchClient.GetNodesAsync(poolID))
+{
+    Console.WriteLine(item.Id);
+}
+```
+
+### Create a Job asynchronously
+A job is a collection of tasks. It manages how computation is performed by its tasks on the compute nodes in a pool.
+
+A job specifies the pool in which the work is to be run. You can create a new pool for each job, or use one pool for many jobs. You can create a pool for each job that is associated with a job schedule, or one pool for all jobs that are associated with a job schedule. For more information see [Jobs and tasks in Azure Batch](https://learn.microsoft.com/azure/batch/jobs-and-tasks).
+
+Use the `CreateJoAsync` method with a `BatchJobCreateContent` instance to create a `BatchJob`. 
+
+```C# Snippet:Batch_Readme_JobCreationAsync
+await _batchClient.CreateJobAsync(new BatchJobCreateContent("jobId", new BatchPoolInfo() { PoolId = "poolName" }));
+```
+
+### Retrieve a job asynchronously
+
+`GetJobAsync` can be used to retrieve a created `BatchJob`
+
+```C# Snippet:Batch_Readme_JobRetreivalAsync
+BatchJob batchJob = await _batchClient.GetJobAsync("jobID");
+Console.WriteLine(batchJob.Id);
+Console.WriteLine(batchJob.State);
+```
+
+### List jobs asynchronously
+`GetJobsAsync` can be used to list all `BatchJob` allocated under a Batch Account
+
+```C# Snippet:Batch_Readme_ListJobsAsync
+await foreach (BatchJob item in _batchClient.GetJobsAsync())
+{
+    Console.WriteLine(item.Id);
+}
+```
+
+### Create a task asynchronously
+A task is a unit of computation that is associated with a job. It runs on a node. Tasks are assigned to a node for execution, or are queued until a node becomes free. Put simply, a task runs one or more programs or scripts on a compute node to perform the work you need done. For more information see [Jobs and tasks in Azure Batch](https://learn.microsoft.com/azure/batch/jobs-and-tasks).
+
+Use the `CreateTaskAsync` method with a `BatchTaskCreateContent` instance to create a `BatchTask`. 
+
+```C# Snippet:Batch_Readme_TaskCreationAsync
+await _batchClient.CreateTaskAsync("jobId", new BatchTaskCreateContent("taskId", $"echo Hello world"));
+```
+
+### Retrieve a task asynchronously
```

Comment:
The goal of the README is to give developers a quick start for the most important champion scenarios for the library.   We generally avoid doing sync/async versions because we want the README to be scenario focused.    

Most libraries use sync in the README for approachability, but that's up to the library.

Reply:
I'd call this non-blocking and at your discretion, but I felt it was worth mentioning.

Reply:
removed the async examples.  I think I saw this pattern in some other teams examples.